frontend/lib/resolution/OverloadSelector.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2025 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/framework/Context.h"
+#include "chpl/framework/ID.h"
+#include "chpl/uast/BuilderResult.h"
+
+namespace chpl {
+namespace resolution {
+
+using BuilderFunc = uast::BuilderResult const&(*)(Context* context, ID typeId, int overloadIdx);
+
+// note: this uses a templated implementation because in modern C++,
+// we should be able to make `dispatch` select the function at
+// compile time. However, presently, some of the older compilers we support
+// do not consider function pointers to be constant expressions, so we
```

Comment:
I'd mention that it's the `==` that breaks down in this comment.

I'd be curious to know if `auto` preserves the "paramness" of the function pointer values when doing `==`. Maybe not but might be fun to try.

Reply:
I tried with `auto` and the error is unchanged.